### PR TITLE
Docs: Update documentation for Elasticsearch 

### LINKED ISSUE #82934: [Elasticsearch] lowercase query operators silently ignored in Lucene queries
### Discussed in https://github.com/grafana/grafana/discussions/63935

<div type='discussions-op-text'>

<sup>Originally posted by **vertebarbe** March  1, 2023</sup>
Hello, I encountered the same problem detailed in #19605.

I understand that this won't be fixed, but there should be a mention about this behavior in the documentation here:
https://grafana.com/docs/grafana/latest/datasources/elasticsearch/query-editor/

I'm sure many would copy the queries straight out of Kibana or are used to write queries like this... which probably resulted in many falsely working panels already.</div>

### LINKED ISSUE #84282: Elasticsearch: Update the last supported version in documentation

### PR DESCRIPTION
**Pull request by ivanahuckova**
This PR updates Elasticsearch documentation:
- removes `and and` (I have found 2 more places where `and and` was in documentation, so I updated that as well)
- updates last supported ES version for Grafana 11 to v `7.17+` based on https://www.elastic.co/support/eol
- adds documentation that users neeed to use uppercase boolean operators AND, OR, NOT and lowercased won't be [arsed by ES

Fixes https://github.com/grafana/grafana/issues/82934
Fixes https://github.com/grafana/grafana/issues/84282

### CODE REVIEW

File: docs/sources/datasources/elasticsearch/query-editor/index.md [hunk 1]
```diff
@@ -24,7 +24,11 @@ weight: 300
 # Elasticsearch query editor
 
 Grafana provides a query editor for Elasticsearch. Elasticsearch queries are in Lucene format.
-See [Lucene query syntax](https://www.elastic.co/guide/en/kibana/current/lucene-query.html) and and [Query string syntax](https://www.elastic.co/guide/en/elasticsearch/reference/8.9/query-dsl-query-string-query.html#query-string-syntax) if you are new to working with Lucene queries in Elasticsearch.
+See [Lucene query syntax](https://www.elastic.co/guide/en/kibana/current/lucene-query.html) and [Query string syntax](https://www.elastic.co/guide/en/elasticsearch/reference/8.9/query-dsl-query-string-query.html#query-string-syntax) if you are new to working with Lucene queries in Elasticsearch.
+
+{{% admonition type="note" %}}
+When composing Lucene queries, ensure that you use uppercase boolean operators: AND, OR, and NOT. Lowercase versions of these operators are not correctly parsed by Elasticsearch.
```
Comment (svennergr) [inline docs/sources/datasources/elasticsearch/query-editor/index.md:30]:
```suggestion
When composing Lucene queries, ensure that you use uppercase boolean operators: `AND`, `OR`, and `NOT`. Lowercase versions of these operators are not supported by the Lucene query language.
```

File: docs/sources/panels-visualizations/query-transform-data/expression-queries/index.md [hunk 2]
```diff
@@ -104,7 +104,7 @@ So if you have numbers with labels like `{host=web01}` in `$A` and another numbe
 - An item with no labels will join to anything.
 - If both `$A` and `$B` each contain only one item (one series, or one number), they will join.
 - If labels are exact math they will join.
-- If labels are a subset of the other, for example and item in `$A` is labeled `{host=A,dc=MIA}` and and item in `$B` is labeled `{host=A}` they will join.
```
Comment (svennergr) [inline docs/sources/panels-visualizations/query-transform-data/expression-queries/index.md:107]: 🚀

File: docs/sources/setup-grafana/configure-security/configure-authentication/saml/index.md [hunk 3]
```diff
@@ -69,7 +69,7 @@ By default, SP-initiated requests are enabled. For instructions on how to enable
    - [`assertion_attribute_email`]({{< relref "../../../configure-grafana/enterprise-configuration#assertion_attribute_email" >}})
    - [`assertion_attribute_name`]({{< relref "../../../configure-grafana/enterprise-configuration#assertion_attribute_name" >}})
    - [`assertion_attribute_groups`]({{< relref "../../../configure-grafana/enterprise-configuration#assertion_attribute_groups" >}})
-1. Save the configuration file and and then restart the Grafana server.
```
Comment (svennergr) [inline docs/sources/setup-grafana/configure-security/configure-authentication/saml/index.md:72]: 🤩

File: docs/sources/datasources/elasticsearch/query-editor/index.md (hunk 1 above)
Comment (svennergr) [reply docs/sources/datasources/elasticsearch/query-editor/index.md:30]:
```suggestion
When composing Lucene queries, ensure that you use uppercase boolean operators: `AND`, `OR`, and `NOT`. Lowercase versions of these operators are not supported by the Lucene query syntax.
```

changed to `Lucene query syntax` as we also use that above.

**Review by svennergr — APPROVED** [review]
One small suggestion, rest LGTM

File: docs/sources/datasources/elasticsearch/query-editor/index.md [hunk 4]
```diff
@@ -24,7 +24,11 @@ weight: 300
 # Elasticsearch query editor
 
 Grafana provides a query editor for Elasticsearch. Elasticsearch queries are in Lucene format.
-See [Lucene query syntax](https://www.elastic.co/guide/en/kibana/current/lucene-query.html) and and [Query string syntax](https://www.elastic.co/guide/en/elasticsearch/reference/8.9/query-dsl-query-string-query.html#query-string-syntax) if you are new to working with Lucene queries in Elasticsearch.
+See [Lucene query syntax](https://www.elastic.co/guide/en/kibana/current/lucene-query.html) and [Query string syntax](https://www.elastic.co/guide/en/elasticsearch/reference/8.9/query-dsl-query-string-query.html#query-string-syntax) if you are new to working with Lucene queries in Elasticsearch.
+
+{{% admonition type="note" %}}
+When composing Lucene queries, ensure that you use uppercase boolean operators: `AND`, `OR`, and `NOT`. Lowercase versions of these operators are not supported by the Lucene query language.
```
Comment (ivanahuckova) [inline docs/sources/datasources/elasticsearch/query-editor/index.md:30]:
```suggestion
When composing Lucene queries, ensure that you use uppercase boolean operators: `AND`, `OR`, and `NOT`. Lowercase versions of these operators are not supported by the Lucene query syntax.
```

**Review by jdbaldry — COMMENTED** [review]
Nice fixes, thank you!

**Pull request merged**